### DISCUSSION TITLE
Exporting raster from rioxarray causes resolution loss

### DISCUSSION THREAD

**Discussion by avery-b** [corteva/rioxarray#470] [Q&A]
I'm working on exporting raster files from NetCDFs, specifically the [Daymet](https://daac.ornl.gov/cgi-bin/dsviewer.pl?ds_id=1840) dataset, using rioxarray and rasterio and have been having difficulty with output raster resolution loss. Would greatly appreciate any guidance here!

Here is my code for exporting the 'day length' field from a NetCDF file:

**Loading data into a rioxarray Dataset**
```
import rioxarray
import xarray
import matplotlib.pyplot as plt

# Load NetCDF with rioxarray
path = '.../daymet_v4_daily_na_dayl_2016.nc'
dL = rioxarray.open_rasterio(path)[0]

# Get rid of missing data
dL_masked = dL.dayl.where(dL.dayl != dL.dayl.attrs['missing_value'])

# Assign crs for North America Lambert Conformal Conic projection
dL_masked.rio.write_crs("esri:102009", inplace=True)

# Plot one day from the dataset
dL_masked[150,:,:].plot()
```
![image](https://user-images.githubusercontent.com/27908596/153365517-3ad5c9a3-b7d3-4667-88a2-f6b175f72557.png)

**Exporting to raster and plotting**

import rasterio
from rasterio.plot import show

dL_masked[150,:,:].rio.to_raster('dayLength.tif')
raster=rasterio.open('dayLength.tif')
show(raster)

![image](https://user-images.githubusercontent.com/27908596/153365714-4e54f15b-21f9-43c4-9ff5-f1a49cd8c37d.png)

**Comment by avery-b** [comment]
Also, I'm using Python 3.9.10, rioxarray 0.9.1, and rasterio 1.2.6

**Comment by snowman2** [comment]
Can you give more details about what is missing?

**Reply by avery-b** [reply]
Sure, as you can see in the second image there is less resolution in the raster than the NetCDF dataset and there is information loss in coastal areas like florida and alaska. Is there a way to use the rio.to_raster function without this loss of resolution and information? I couldn't find any relevant parameters in the documentation.

**Reply by snowman2** [reply]
Mind providing the resolution, dimensions, and transform of the raster before export and after export?

**Reply by avery-b** [reply]
Sure thing, heres the info. They look similar so I'm not sure where the issue arises

**Original NetCDF:**
resolution: (1000.0, -1000.0)
transform:
Affine(1000.0, 0.0, -4560750.0,
       0.0, -1000.0, 4984500.0)

**Output Raster:**
resolution: (1000.0, -1000.0)
transform: 
| 1000.00, 0.00,-4560750.00|
| 0.00,-1000.00, 4984500.00|
| 0.00, 0.00, 1.00|

**Reply by avery-b** [reply]
Here are the files by the way:
https://drive.google.com/drive/folders/19xNOGAM5kor_pORjjCD2C6xguxPZv2Xa?usp=sharing

Thanks!

**Reply by snowman2** [reply]
Sounds to me like this isn't an issue with rioxarray as much as a difference in the methods for plotting the raster.

I suspect if you do this, it will resemble the original plot:
```python
import rioxarray
rds = rioxarray.open_rasterio('dayLength.tif', mask_and_scale=True)
rds.squeeze().plot()
```

**Reply by avery-b** [reply]
Yes thank you! That seems to do the trick. What are the squeeze and mask_and_scale doing here?

**Reply by snowman2** [reply]
- `mask_and_scale`: https://corteva.github.io/rioxarray/stable/rioxarray.html#rioxarray-open-rasterio
- `squeeze`: https://xarray.pydata.org/en/stable/generated/xarray.DataArray.squeeze.html

**Reply by avery-b** [reply]
Quick question @snowman2. I'm noticing that exporting with rio.to_raster shifts the output so that it doesn't line up with the original raster read directly from the NetCDF file. Is there a way to retain the exact Affine and projection information in the output tif file?

![image](https://user-images.githubusercontent.com/27908596/153612949-45706de0-0887-440b-b102-93bc97ee0408.png)

**Reply by snowman2** [reply]
From your previous post, it has the same transform in the raster as it has in the netCDF file. What are you using to view the files?

**Reply by avery-b** [reply]
QGIS

**Reply by snowman2** [reply]
What does it look like with a basemap?

**Reply by avery-b** [reply]
This is the exported .tiff file using rioxarray:
![image](https://user-images.githubusercontent.com/27908596/153867690-cbc05d8a-ec48-4475-af63-5ad048d7d320.png)

This is the original NetCDF file:
![image](https://user-images.githubusercontent.com/27908596/153867820-32fc660d-6ef5-4262-a638-e425f32d4fc4.png)

**Reply by snowman2** [reply]
I tried to download the files and got a 404. If it is plotting with QGIS, I am wondering if there is an original CRS on the netCDF originated dataset. After opening the `netCDF` file, are you able to get a CRS object by doing `dL.rio.crs`?

**Reply by avery-b** [reply]
Reuploaded the files here: https://drive.google.com/drive/folders/1uDxtknGk-gIZZCNNLyHBjbyGswezIUXx?usp=sharing

I'm able to get the crs object with the function dL.dayl.rio.crs:

CRS.from_wkt('PROJCS["unnamed",GEOGCS["unknown",DATUM["unnamed",SPHEROID["Spheroid",6378137,298.257223563]],PRIMEM["Greenwich",0],UNIT["degree",0.0174532925199433,AUTHORITY["EPSG","9122"]]],PROJECTION["Lambert_Conformal_Conic_2SP"],PARAMETER["latitude_of_origin",42.5],PARAMETER["central_meridian",-100],PARAMETER["standard_parallel_1",25],PARAMETER["standard_parallel_2",60],PARAMETER["false_easting",0],PARAMETER["false_northing",0],UNIT["metre",1,AUTHORITY["EPSG","9001"]],AXIS["Easting",EAST],AXIS["Northing",NORTH]]')

**Reply by avery-b** [reply]
Possibly relevant, when I open the file with dL = rioxarray.open_rasterio(dayLength, mask_and_scale=True)[0], I get this warning:

/opt/anaconda3/envs/netcdf/lib/python3.9/site-packages/rasterio/__init__.py:220: NotGeoreferencedWarning: Dataset has no geotransform, gcps, or rpcs. The identity matrix be returned.
  s = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)

**Reply by snowman2** [reply]
> I'm able to get the crs object with the function dL.dayl.rio.crs:

What happens if you write using `rio.to_raster` without setting the CRS to the ESRI code? Does it appear correct then?

**Reply by avery-b** [reply]
Seems to do the trick. Thanks again man, appreciate the help.
